src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -311,7 +320,22 @@ void validateStudyDatasetAndInstances(
 
 	@Override
 	public List<ObservationUnitRow> getObservationUnitRows(
-		final int studyId, final int datasetId, final ObservationUnitsSearchDTO searchDTO) {
+		final int studyId, final int datasetId, final ObservationUnitsSearchDTO searchDTO, final Pageable pageable) {
+
+		Pageable convertedPageable = null;
+		if (pageable != null && pageable.getSort() != null) {
+			final Iterator<Sort.Order> iterator = pageable.getSort().iterator();
+			if (iterator.hasNext()) {
+				// Convert the sort property name from termid to actual term name.
```

Comment:
1. Why did you move this logic to lookup term name by term id from Middleware to BMSAPI?
2. Another note is maybe we can find a way to optimize by getting all variables IDS one-off

Reply:
I moved that logic from Middleware (fillSearchDto) to ensure that the Pageable's order by value is already converted to the proper term name before passing it to the middleware method. Pageable is an interface so we cannot transform/overwrite that inside the middleware method.